tf-psa-crypto/drivers/builtin/src/ccm.c

```diff
@@ -172,6 +172,7 @@ static int ccm_calculate_first_block_if_ready(mbedtls_ccm_context *ctx)
     if (ctx->tag_len == 0) {
         if (ctx->mode == MBEDTLS_CCM_STAR_ENCRYPT || ctx->mode == MBEDTLS_CCM_STAR_DECRYPT) {
             ctx->plaintext_len = 0;
+            return ret;
```

Comment:
At this point `ret` is still `MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED`. And if an error was possible, we'd need to set the state to `CCM_STATE__ERROR`. (Preexisting: we should do that in the bad-mode case.)
```suggestion
            return 0;
```